index.js

```diff
@@ -250,29 +271,51 @@ class K8sExecutor extends Executor {
     /**
      * Starts a k8s build
      * @method start
-     * @param  {Object}   config            A configuration object
-     * @param  {Integer}  config.buildId    ID for the build
-     * @param  {String}   config.container  Container for the build to run in
-     * @param  {String}   config.token      JWT for the Build
+     * @param  {Object}   config                A configuration object
+     * @param  {Integer}  config.buildId        ID for the build
+     * @param  {Integer}  config.pipeline.id    pipelineId for the build
+     * @param  {Integer}  config.jobId          jobId for the build
+     * @param  {Integer}  config.eventId        eventId for the build
+     * @param  {String}   config.container      Container for the build to run in
+     * @param  {String}   config.token          JWT for the Build
+     * @param  {String}   config.jobName        jobName for the build
      * @return {Promise}
      */
     _start(config) {
-        const { buildId, container, token } = config;
-        const random = randomstring.generate({
-            length: 5,
-            charset: 'alphanumeric',
-            capitalization: 'lowercase'
-        });
+        const { buildId, eventId, container, token } = config;
+        let jobId = hoek.reach(config, 'jobId', { default: '' });
+        const pipelineId = hoek.reach(config, 'pipeline.id', { default: '' });
+        const jobName = hoek.reach(config, 'jobName', { default: '' });
         const annotations = this.parseAnnotations(
             hoek.reach(config, 'annotations', { default: {} }));
         const cpuValues = {
+            MAX: this.maxCpu,
             TURBO: this.turboCpu,
             HIGH: this.highCpu,
             LOW: this.lowCpu,
             MICRO: this.microCpu
         };
+
+        // PRs - set pipeline, job cache volume readonly and job cache dir to parent job cache dir
+        const regex = /^PR-([0-9]+)(?::[\w-]+)?$/gi;
```

Comment:
save as a constant?
might be good to have a more descriptive var name too

Reply:
done